editor/plugins/script_text_editor.cpp

```diff
@@ -1598,9 +1598,28 @@ static Node *_find_script_node(Node *p_edited_scene, Node *p_current_node, const
 	return nullptr;
 }
 
-void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from) {
-	const String quote_style = EDITOR_GET("text_editor/completion/use_single_quotes") ? "'" : "\"";
 
+String ScriptTextEditor::_quote_drop_data(const String &str) {
+	// This function prepares a string for being "dropped" into the script editor.
+	// The string can be a resource path, node path or property name.
+
+	const bool using_single_quotes = EDITOR_GET("text_editor/completion/use_single_quotes");
+
+	// Escape any special characters in the string
+	String escaped = str.c_escape();
+
+	// Unescape single quotes if user prefers double-quote delimited strings.
```

Comment:
Should explain the why a bit cleaner, something like:
```suggestion
	// If string is double quoted, there is no need to escape single quotes, so we can remove the extra escaping added above here.
```